### PR TITLE
update dependency and add demo for consul2istio

### PR DESCRIPTION
**Pull request by tanjunchen**
@zhaohuabing

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,15 +1,95 @@
...
+// Client-go does not handle different versions of mergo due to some breaking changes - use the matching version
+replace github.com/imdario/mergo => github.com/imdario/mergo v0.3.5
+
+require (
+	github.com/hashicorp/consul/api v1.8.1
+	google.golang.org/protobuf v1.28.1
+	istio.io/api v0.0.0-20230518153929-d0aebaa77ab8
+	istio.io/client-go v1.16.4-0.20230518154329-f75cb9ff8e52
+	istio.io/istio v0.0.0-20230519000352-ae8d5164776c
+	istio.io/pkg v0.0.0-20221107183613-574f8d141535
```
Comment (tanjunchen) [inline go.mod:17]: 更新为 istio 1.16.x  依赖包

File: README.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# Consul2Istio
+
+[![CI Tests](https://github.com/aeraki-framework/consul2istio/workflows/ci/badge.svg?branch=master)](https://github.com/aeraki-framework/consul2istio/actions?query=branch%3Amaster+event%3Apush+workflow%3A%22ci%22)
+
```
Comment (tanjunchen) [inline README.md:4]: CI Pass @zhaohuabing

File: Makefile [hunk 3]
```diff
@@ -9,22 +9,23 @@ GOBIN?=$(GOPATH)/bin
 # Build parameters
 OUT?=./out
 DOCKER_TMP?=$(OUT)/docker_temp/
-DOCKER_TAG?=aeraki/consul2istio:latest
+DOCKER_TAG?=docker.io/tanjunchen/consul2istio:latest
```
Comment (zhaohuabing) [inline Makefile]: the tag is wrong

File: Makefile [hunk 4]
```diff
@@ -9,22 +9,23 @@ GOBIN?=$(GOPATH)/bin
...
 BINARY_NAME?=$(OUT)/consul2istio
 BINARY_NAME_DARWIN?=$(BINARY_NAME)-darwin
 MAIN_PATH_CONSUL_MCP=./cmd/consul2istio/main.go
 
 build:
-	CGO_ENABLED=0 GOOS=linux  $(GOBUILD) -o $(BINARY_NAME) $(MAIN_PATH_CONSUL_MCP)
+	CGO_ENABLED=0 GOARCH=amd64 GOOS=linux  $(GOBUILD) -o $(BINARY_NAME) $(MAIN_PATH_CONSUL_MCP)
```
Comment (zhaohuabing) [inline Makefile]: the default goarch is supposed to be x86

File: Makefile (hunk 3 above)
Comment (tanjunchen) [reply Makefile]: Where should I push the image? ghcr.io/aeraki-mesh？

File: Makefile (hunk 4 above)
Comment (tanjunchen) [reply Makefile]: I make this parameter configurable.

File: Makefile (hunk 3 above)
Comment (zhaohuabing) [reply Makefile]: Yes, we use github packages for releasing aeraki mesh images.

**Review by zhaohuabing — APPROVED** [review]
LGTM.

Thanks!

File: Makefile (hunk 3 above)
Comment (tanjunchen) [reply Makefile]: updated.

**Pull request merged**